blog/_posts/2022-07-18-secure-your-dependencies-on-github.md

```diff
@@ -0,0 +1,139 @@
+ 	    working-directory: ‘./’
+```
+
+This default definition should work in many sbt builds out-of-the-box.
+The action will check that the working directory contains an sbt project, it will install an sbt plugin, and run a command, added by that plugin, that extracts all resolved dependencies from all Scala projects on all Scala versions.
+
+If you need to exclude a project on a Scala version, because sbt cannot resolve its dependencies, you can use the `modules-ignore` input.
+You can learn more about this in the [documentation page](https://github.com/marketplace/actions/sbt-dependency-submission) of the action.
+
+## Scala Steward and GitHub Dependabot
+
+[Scala Steward](https://github.com/scala-steward-org/scala-steward) is a tool that helps you keep the dependencies of your project up-to-date by opening pull requests on GitHub (and other hosting services).
+It is used in more than a thousand open-source repositories and many proprietary ones and contributes to a large extent to the security of the Scala ecosystem.
+
+Dependabot and Scala Steward can be used as complementary tools.
+Scala Steward, as a preventive tool, can help you keep your dependencies up-to-date, which reduces the risk of security vulnerabilities.
+Dependabot, as a monitoring tool, can notify you when a vulnerability is found, so that you can act quickly.
+
+## Other supported build tools
+
+You can use Dependabot with other Scala-compatible build tools.
+
+### Maven
+
+Dependabot supports Maven through static analysis of POM files.
+You can enable the Dependency graph view and Dependabot in the `Settings > Code security and analysis` page.
+
+### Mill
+The [mill-dependency-submission][mill-dependency-submisison] action, developed by Chris Kipp, can extract the dependencies of a Mill build.
+
+To install it:
+ - Enable the `Dependency Graph` view, and optionally the `Dependabot alerts`, in the `Settings > Code security and analysis` page of your repository.
+ - Add this workflow in you repository:
+
+```yaml
+# .github/workflows/dependency-graph.yml
+name: Update Dependency Graph
+
+on:
+  push:
+    branches:
+      - main # default branch of the project
+
+# The API requires write permission on the repository to submit dependencies
+permissions:
+  contents: write
+
+jobs:
+  update-graph:
+    name: Update Dependency Graph
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: coursier/cache-action@v6
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+      - uses: ckipp01/mill-dependency-submission@v1
+```
+### Gradle
+
+Gradle users can use the [gradle-dependency-submission][gradle-dependency-submisison] action.
+
+## Summary
+
+In Github, it is now possible to use pre-made actions, like [sbt-dependency-submisison][sbt-dependency-submisison] or [mill-dependency-submisison][mill-dependency-submisison], to extract full graphs of dependencies at build time.
```

Comment:
There's a typo in "submission" as submisison both in the text and the Markdown link name.